agent/04_agent_configure.sh

```diff
@@ -37,40 +38,47 @@ function get_static_ips_and_macs() {
     for (( i=0; i<${static_ips}; i++ ))
     do
         ip=${base_ip}+${i}
-        FLEETING_NODES_IPS+=($(nth_ip ${external_subnet} ${ip}))
+        AGENT_NODES_IPS+=($(nth_ip ${external_subnet} ${ip}))
 
         if [[ $i < $NUM_MASTERS ]]; then
-            FLEETING_NODES_HOSTNAMES+=($(printf ${MASTER_HOSTNAME_FORMAT} ${i}))
+            AGENT_NODES_HOSTNAMES+=($(printf ${MASTER_HOSTNAME_FORMAT} ${i}))
             cluster_name=${CLUSTER_NAME}_master_${i}
         else
 	    worker_num=$((${i}-$NUM_MASTERS))
-            FLEETING_NODES_HOSTNAMES+=($(printf ${WORKER_HOSTNAME_FORMAT} ${worker_num}))
+            AGENT_NODES_HOSTNAMES+=($(printf ${WORKER_HOSTNAME_FORMAT} ${worker_num}))
             cluster_name=${CLUSTER_NAME}_worker_${worker_num}
         fi
 
-        # Add a DNS entry for this hostname
-        sudo virsh net-update ${BAREMETAL_NETWORK_NAME} add dns-host  "<host ip='${FLEETING_NODES_IPS[i]}'> <hostname>${FLEETING_NODES_HOSTNAMES[i]}</hostname> </host>"  --live --config
+        # Add a DNS entry for this hostname if it's not already defined
+        if ! $(sudo virsh net-dumpxml ${BAREMETAL_NETWORK_NAME} | xmllint --xpath "//dns/host[@ip = '${AGENT_NODES_IPS[i]}']" - &> /dev/null); then
+          sudo virsh net-update ${BAREMETAL_NETWORK_NAME} add dns-host  "<host ip='${AGENT_NODES_IPS[i]}'> <hostname>${AGENT_NODES_HOSTNAMES[i]}</hostname> </host>"  --live --config
+        fi
 
         # Get the generated mac addresses
-        FLEETING_NODES_MACS+=($(sudo virsh dumpxml $cluster_name | xmllint --xpath "string(//interface[descendant::source[@bridge = '${BAREMETAL_NETWORK_NAME}']]/mac/@address)" -))
+        AGENT_NODES_MACS+=($(sudo virsh dumpxml $cluster_name | xmllint --xpath "string(//interface[descendant::source[@bridge = '${BAREMETAL_NETWORK_NAME}']]/mac/@address)" -))
     done
 }
 
-function generate_fleeting_manifests() {
+function generate_cluster_manifests() {
 
-    mkdir -p ${FLEETING_MANIFESTS_PATH}
+  # To be change in ${OCP_DIR}/cluster-manifests
+  MANIFESTS_PATH="${OCP_DIR}/manifests"
 
-    if [[ "$IP_STACK" = "v4" ]]; then
-      CLUSTER_NETWORK=${CLUSTER_SUBNET_V4}
-      SERVICE_NETWORK=${SERVICE_SUBNET_V4}
-      CLUSTER_HOST_PREFIX=${CLUSTER_HOST_PREFIX_V4}
-    elif [[ "$IP_STACK" = "v6" ]]; then
-      CLUSTER_NETWORK=${CLUSTER_SUBNET_V6}
-      SERVICE_NETWORK=${SERVICE_SUBNET_V6}
-      CLUSTER_HOST_PREFIX=${CLUSTER_HOST_PREFIX_V6}
-    fi
+  if [ ! -d ${MANIFESTS_PATH} ]; then
+    mkdir -p ${MANIFESTS_PATH}
```

Comment:
this is fine, just the -p option does not error out if the destination exists, so technically you won't need the if clause